backend/lib/deal-submitter.js

```diff
@@ -0,0 +1,115 @@
+/** @import {PgPool, Queryable} from '@filecoin-station/deal-observer-db' */
+import Cursor from 'pg-cursor'
+
+/**
+ * Find deals that are eligible to be submitted to a spark api and submits them.
+ *
+ * @param {PgPool} pgPool
+ * @param {string} sparkApiBaseURL
+ * @param {string} dealIngestionAccessToken
+ * @param {number} batchSize
+ * @param {(sparkApiBaseURL: string, dealIngestionAccessToken: string, eligibleDeals: Array) => Promise<void>} submitEligibleDealsFn
+ */
+export const findAndSubmitEligibleDeals = async (pgPool, sparkApiBaseURL, dealIngestionAccessToken, batchSize, submitEligibleDealsFn) => {
+  for await (const eligibleDeals of findEligibleDeals(pgPool, batchSize)) {
+    const formattedEligibleDeals = formatEligibleDeals(eligibleDeals)
+    await submitEligibleDealsFn(sparkApiBaseURL, dealIngestionAccessToken, formattedEligibleDeals)
+    await markEligibleDealsSubmitted(pgPool, eligibleDeals)
+  }
+}
+
+/**
+ * Find deals that are eligible to be submitted to a spark api.
```

Comment:
```suggestion
 * Find deals that are eligible to be submitted to spark api.
```

There is only one spark api